File: frontEnd/train_truck/lib/Models/Stop.dart
class Stop {
  int? stopId;
  String? stopCode;
  String? stopName;
  String? stopDesc;
  double? stopLat;
  double? stopLon;
  String? zoneId;
  String? stopUrl;
  String? locationType;
  String? parentStation;

  Stop(
      {this.stopId,
      this.stopCode,
      this.stopName,
      this.stopDesc,
      this.stopLat,
      this.stopLon,
      this.zoneId,
      this.stopUrl,
      this.locationType,
      this.parentStation});

  Stop.fromJson(dynamic json) {
    stopCode = json['stop_code'];
    stopName = json['stop_name'];
    stopDesc = json['stop_desc'];
    stopLat = json['stop_lat'];
    stopLon = json['stop_lon'];
    zoneId = json['zone_id'];
    stopUrl = json['stop_url'];
    locationType = json['location_type'];
    parentStation = json['parent_station'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['stop_id'] = this.stopId;
    data['stop_code'] = this.stopCode;
    data['stop_name'] = this.stopName;
    data['stop_desc'] = this.stopDesc;
    data['stop_lat'] = this.stopLat;
    data['stop_lon'] = this.stopLon;
    data['zone_id'] = this.zoneId;
    data['stop_url'] = this.stopUrl;
    data['location_type'] = this.locationType;
    data['parent_station'] = this.parentStation;
    return data;
  }
}

File: backend/trainTruck/src/main/java/com/trainTruck/demo/Model/Stop.java
package com.trainTruck.demo.Model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="stops")
public class Stop {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int stop_id;
	
	@Column(name="stop_code")
	private String stop_code;
	
	private String stop_name;
	
	private String stop_desc;
	
	@Column(name="stop_lat")
	private double stop_lat=0.0;
	
	@Column(name="stop_lon")
	private double stop_lon=0.0;
	
	private String zone_id;
	private String stop_url;
	private String location_type;
	private String parent_station;
	


}